naucse/routes.py

```diff
@@ -51,6 +54,15 @@ def lesson_url(lesson, page='index', solution=None):
     return url_for('lesson', lesson=lesson, page=page, solution=solution)
 
 
+@template_function
+def session_url(run, session, page='front'):
+    session_page = "runs/" + run + "/sessions/" + session + '/' + page + ".md"
+    if (os.path.exists(session_page)):
+        return url_for('session_page', run=run, session=session, page=page)
+    else:
+        return url_for('run', run=run)
+    
+
```

Comment:
This breaks the “model” abstraction.

Code in `routes.py` should not interact directly with the file system. It should be easy to point the website to a different model – a different directory conaining `lessons/`, `runs/`, etc.
It should be possible to switch to storing the content in a database instead of on disk.

Please add any code that reads/renders session content to `models.py`.